### PR TITLE
Update to use generic PKCS#11 PSA layer

### PR DESCRIPTION
**Pull request by CharleyChu**
Signed-off-by: Charley Chu <haoc@cypress.com>

Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/CMakeLists.txt [hunk 1]
```diff
@@ -100,6 +100,34 @@ else()
     )
 endif()
 
+afr_module(NAME pkcs11_psa INTERFACE)
```
Comment (lundinc2) [inline libraries/abstractions/pkcs11/CMakeLists.txt:103]: Could you wrap this module in an else if block following the pattern above, using `PSA_IN_USE` for example as a variable, so this is not included in all the platforms

Comment (CharleyChu) [reply libraries/abstractions/pkcs11/CMakeLists.txt:103]: Done.

**Comment by CharleyChu** [comment]
Fixed

**Comment by CharleyChu** [comment]
Should be fixed with latest update. Thanks!

**Comment by lundinc2** [comment]
Hi @CharleyChu,

Thank you for applying those changes. I ran a regression, and our tests don't seem to pass for mtb, but they do for CMake, can you run them locally and address the failures?

Thanks,

Carl

**Comment by RaymondNgun** [comment]
@lundinc2 (giving this thread a bump)

**Comment by lundinc2** [comment]
https://github.com/aws/amazon-freertos/blob/dd97ccc7e621d8de8cf579854f4672c92e597c13/libraries/abstractions/pkcs11/test/iot_test_pkcs11.c#L352 this test group, with the following results
```
.1 8270 [Tmr Svc] Wi-Fi Connected to AP. Creating tasks which use network...
2 8270 [Tmr Svc] IP Address acquired 192.168.35.148
3 9125 [Tmr Svc] ERROR: Find object operation already in progress. 
4 9125 [Tmr Svc] ERROR: Find object operation already in progress. 
5 9125 [Tmr Svc] Warning: could not clean-up old crypto objects. 144 
.---------STARTING TESTS---------
UUID: _uuid_235bd35eabaa4c308c2acdec71d6fcbf
TEST(Full_PKCS11_EC, AFQP_GenerateKeyPair)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:1843::FAIL: Expected 0 Was 6. Generating EC key pair failed.
.6 15028 [RunTests_] ERROR: Find object operation already in progress. 
7 15028 [RunTests_] ERROR: Find object operation already in progress. 
TEST(Full_PKCS11_EC, AFQP_CreateObject)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:468::FAIL: Expected 0 Was 144. Found an object after deleting it.\r\n
TEST(Full_PKCS11_EC, AFQP_FindObject)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_GetAttributeValue)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_Sign)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_Verify)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_FindObjectMultiThread)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_GetAttributeValueMultiThread)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
TEST(Full_PKCS11_EC, AFQP_SignVerifyMultiThread)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:600::FAIL: Expected 0 Was 6. Failed to find private key after closing and reopening a session.
8 16639 [RunTests_] ERROR: Find object operation already in progress. 
9 16639 [RunTests_] ERROR: Find object operation already in progress. 
TEST(Full_PKCS11_EC, AFQP_DestroyObject)../../../../../libraries/abstractions/pkcs11/test/iot_test_pkcs11.c:468::FAIL: Expected 0 Was 144. Found a
 ob
```

My guess is something is wrong with the build, as the same board passed this test when using CMake

**Comment by lundinc2** [comment]
@CharleyChu the code in the master branch didn't show this issue

**Comment by CharleyChu** [comment]
@lundinc2 So, you are running IDT tests with MTB, right?

**Comment by lundinc2** [comment]
@CharleyChu It's the same tests as IDT, but I am running them with this project https://github.com/aws/amazon-freertos/tree/master/projects/cypress/CY8CKIT_064S0S2_4343W/mtb/aws_tests.

**Comment by CharleyChu** [comment]
@lundinc2 Got it. Thanks!

**Comment by lundinc2** [comment]
> @lundinc2 Got it. Thanks!

No problem! Let me know when it's fixed, and I'll go ahead and merge this!

Thanks,

Carl

**Comment by CharleyChu** [comment]
@lundinc2 Tested locally, the FullPKCS11_ECC test passes with MTB build.

Thanks!
Charley

**Comment by lundinc2** [comment]
@CharleyChu that did it! Thanks for the fast work!

**Review by lundinc2 — APPROVED** [review]

**Comment by CharleyChu** [comment]
@lundinc2 Good to hear that. Thanks!

**Review by n9wxu — APPROVED** [review]

**Pull request merged**